File: app/src/main/java/com/casasw/bankapp/LoginRouter.java
package com.casasw.bankapp;

import android.content.Intent;

import java.lang.ref.WeakReference;


interface LoginRouterInput {

    void passDataToNextScene(LoginViewModel viewModel, Intent intent);
}

public class LoginRouter implements LoginRouterInput{

    public static String TAG = LoginRouter.class.getSimpleName();
    WeakReference<LoginActivity> activity;



    @Override
    public void passDataToNextScene(LoginViewModel viewModel, Intent intent) {
        intent.putExtra("EXTRA_LOGIN_MODEL", viewModel);
        activity.get().startActivity(intent);
    }


}

File: app/src/main/java/dev/vitorpacheco/solutis/bankapp/loginScreen/LoginRouter.kt
package dev.vitorpacheco.solutis.bankapp.loginScreen

import android.content.Intent
import java.lang.ref.WeakReference

internal interface LoginRouterInput {
    fun navigateToStatements(account: UserAccount): Intent
}

class LoginRouter : LoginRouterInput {

    lateinit var activity: WeakReference<LoginActivity>

    override fun navigateToStatements(account: UserAccount): Intent {
//        val intent = Intent(activity.get(), StatementsActivity::class.java)
//        intent.putExtra("account", account)
//
//        return intent
        return Intent()
    }

    companion object {
        var TAG = LoginRouter::class.java.simpleName
    }

}